script.tag-generator/main.py

```diff
@@ -370,152 +449,191 @@ def writestanduptags(comiclist, Medialist, newwikitag):
+                except Exception as e:
                     dialog = xbmcgui.Dialog()
-                    ok = dialog.ok("Tag Generator", _getstr(30925)+str(e.message))
-                    sys.exit(_getstr(30927))
+                    ok = dialog.ok("Tag Generator", _getstr(30925))
+                    xbmc.log(msg="Unable to retrieve Trakt oauth token." + str(e), level=xbmc.LOGERROR)
+                    sys.exit(0)
             else:
-                sys.exit(_getstr(30927))
+                xbmc.log(msg="TAG-GEN: Manual Trakt Init arg received, exiting after execution.", level=xbmc.LOGNOTICE)
+                sys.exit(0)
         elif sys.argv[1] == "trakt":
             if len(c_trakt_token) != 64:
                 dialog = xbmcgui.Dialog()
                 ok = dialog.ok("Tag Generator", _getstr(30925))
-                xbmc.log(msg= _getstr(30925),level=xbmc.LOGNOTICE)
-                sys.exit(_getstr(30925))
+                xbmc.log(msg="TAG-GEN: Unable to retrieve Trakt oauth token.", level=xbmc.LOGERROR)
+                sys.exit(1)
             pDialog = xbmcgui.DialogProgress()
-            ret = pDialog.create("Tag Generator", _getstr(30064))
-            xbmc.log(msg= _getstr(30070),level=xbmc.LOGNOTICE)
-            Medialist = getxbmcdb()
+            ret = pDialog.create("Tag Generator", _getstr(30027))
+            xbmc.log(msg="TAG-GEN: Starting Trakt writing.", level=xbmc.LOGNOTICE)
+            medialist = getxbmcdb()
             i = 0
             for this_trakt_list in trakt_lists:
                 this_trakt_tag = trakt_tags[i]
                 this_trakt_user = trakt_users[i]
                 try:
                     trakt_movies = get_trakt_movies(this_trakt_user, this_trakt_list, c_trakt_token)
-                    moviecount += write_trakt_tags(trakt_movies, Medialist, this_trakt_tag)
+                    moviecount += write_trakt_tags(trakt_movies, medialist, this_trakt_tag)
                 except:
-                    xbmc.log(msg= _getstr(30928),level=xbmc.LOGERROR)
-                i+=1
+                    xbmc.log(msg="TAG-GEN: Could not retrieve movies from Trakt API.", level=xbmc.LOGERROR)
+                i += 1
+            dialog = xbmcgui.Dialog()
+            ok = dialog.ok("Tag Generator", _getstr(30031) + str(moviecount) + _getstr(30003))
+            xbmc.log(msg="TAG-GEN: Manual arg received, exiting after single execution.", level=xbmc.LOGNOTICE)
+            sys.exit(0)
+        elif sys.argv[1] == "awards":
+            if not (c_oscarwinners or c_oscarnominees):
+                dialog = xbmcgui.Dialog()
+                ok = dialog.ok("Tag Generator", _getstr(30043))
+                xbmc.log(msg="TAG-GEN: No awards category chosen", level=xbmc.LOGERROR)
+                sys.exit(1)
+            pDialog = xbmcgui.DialogProgress()
+            ret = pDialog.create("Tag Generator", _getstr(30027))
+            xbmc.log(msg="TAG-GEN: Starting awards tag writing.", level=xbmc.LOGNOTICE)
```

Comment:
log spam

Reply:
This one remains; it's a large activity that consumes resources and should not be invisible by default.